app/src/main/java/com/android/solvit/provider/ui/profile/ProviderProfile.kt

```diff
@@ -71,508 +69,258 @@ import com.android.solvit.shared.ui.navigation.Screen
-                .wrapContentHeight()
-                .padding(8.dp)
-                .weight(1f),
+            Modifier.fillMaxSize()
+                .padding(paddingValues)
+                .padding(horizontal = 16.dp)
+                .verticalScroll(rememberScrollState())
+                .testTag("ProfileContent"),
         horizontalAlignment = Alignment.CenterHorizontally,
-        verticalArrangement = Arrangement.Top) {
-          Column(
-              modifier = Modifier.align(Alignment.Start),
-              horizontalAlignment = Alignment.Start,
-              verticalArrangement = Arrangement.Bottom) {
-                Box {
-                  IconButton(
-                      onClick = { navigationActions.goBack() },
-                      modifier = Modifier.testTag("backButton")) {
-                        Icon(
-                            Icons.AutoMirrored.Filled.ArrowBack,
-                            contentDescription = "GoBackButton",
-                            modifier = Modifier.size(24.dp),
-                            tint = colorScheme.error)
-                      }
-                }
+        verticalArrangement = Arrangement.spacedBy(16.dp)) {
+
+          // profile info card
+          ProfileInfoCard(
+              fullName = provider?.name ?: "",
+              email = provider?.companyName ?: "",
+              imageUrl = provider?.imageUrl ?: "",
+              onEdit = { navigationActions.navigateTo(Screen.PROVIDER_MODIFY_PROFILE) })
+
+          // additional infos cards
+          Card(
+              modifier = Modifier.fillMaxWidth().testTag("AdditionalInfosCard"),
+              colors = CardDefaults.cardColors(containerColor = colorScheme.background),
+              elevation = CardDefaults.cardElevation(defaultElevation = 4.dp)) {
+                Column(
+                    verticalArrangement = Arrangement.spacedBy(8.dp),
+                    modifier = Modifier.padding(16.dp)) {
+                      ProviderItem(
+                          icon = Icons.Default.Build,
+                          text = provider?.service?.name ?: "",
+                          tag = "ServiceItem")
+                      ProviderItem(
+                          icon = Icons.Default.LocationOn,
+                          text = provider?.location?.name ?: "",
+                          tag = "LocationItem")
+                      ProviderItem(
+                          icon = Icons.Default.Info,
+                          text = provider?.description ?: "",
+                          tag = "DescriptionItem",
+                          maxLines = 3)
+                      ProviderItem(
+                          icon = Icons.Default.ThumbUp,
+                          drawable = R.drawable.language_icon,
+                          text = provider?.languages?.joinToString { it.name.lowercase() } ?: "",
+                          tag = "LanguagesItem")
```

Comment:
Same remark about the formatting of the languages, capitalizing the first letter would improve the display, in my opinion.

Reply:
I'll do just that